### PR TITLE
#157469427 Design asset category cards

### PR DESCRIPTION
**Pull request by Billmike**
#### What does this PR do?
Setup the component for the display of asset category cards

### Description of task to be completed?
- create dashboard component
- create globals, mixins and variables scss files
- import DinPro font file
- write test for Assest category cards

#### How should this be manually tested?
- After cloning the repo, run `yarn install` to install the necessary packages
- Run `yarn start:dev` to start the webpack dev-server on `port 3000`.
- Open web browser to `localhost:3000` to see the dashboard component

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#157469427](https://www.pivotaltracker.com/story/show/157469427)

#### Screenshots
screen shot 2018-05-28 at 3 47 48 pm

### CODE REVIEW

**Comment by andela-kakpobome** [comment]
Your commit messages should be in all lowercase

File: config/webpack.config.js [hunk 1]
```diff
@@ -9,10 +9,13 @@ const {
...
 module.exports = {
-  entry: path.join(__dirname, '..', 'src', 'index.tsx'),
+  entry: {
+    bundle: path.join(__dirname, '..', 'src', 'index.tsx'),
+    styleGlobals: path.join(__dirname, '..', 'src/assets/scss/globals.scss')
+  },
   output: {
     path: path.join(__dirname, '..', 'dist'),
-    filename: 'bundle.min.js',
+    filename: '[name].min.js',
```
Comment (andela-kakpobome) [inline config/webpack.config.js]: Let's change this to `[name].js` and then in the `webpack.prod.js`, have it be `[name].min.js`

File: config/webpack.dev.js [hunk 2]
```diff
@@ -22,5 +26,12 @@ module.exports = merge(config, {
       ignored: /node_modules/
     }
   },
-  plugins: [namedModulesPlugin, hotModuleReplacementPlugin],
+  plugins: [
+    namedModulesPlugin,
+    hotModuleReplacementPlugin,
+    new MiniCssExtractPlugin({
```
Comment (andela-kakpobome) [inline config/webpack.dev.js]: Let's have this in the general `wepack.config.js`, and abstract the implementation in the `webpack.plugins` file

File: config/webpack.dev.js [hunk 3]
```diff
@@ -22,5 +26,12 @@ module.exports = merge(config, {
...
+      filename: "[name].css",
+      chunkFilename: "[id].css"
```
Comment (andela-kakpobome) [inline config/webpack.dev.js]: Let's stick with just `filename`, an d lose this line.

File: src/assets/scss/variables.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+$andela-blue: #3359db;
+$light-gray: #323c47;
```
Comment (andela-kakpobome) [inline src/assets/scss/variables.scss]: Capitalize the `hexes`

File: src/assets/scss/variables.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+$large: 24px;
+$mid-size: 20px;
```
Comment (andela-kakpobome) [inline src/assets/scss/variables.scss]: 1. $large and $mid-size are not proper variable names. I can't even tell what they are meant for and hence can't suggest an alternative.

2. Think through and have more descriptive names.

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 6]
```diff
@@ -0,0 +1,91 @@
+@import '../../assets/scss/variables';
+
+.grid-item {
+  background-color: #fff;
+  border-radius: 6px;
+  box-shadow: 0 2px 9px 0 rgba(0, 0, 0, 0.05);
+  cursor: pointer;
+  float: left;
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: 1. As I've repeated several times, we are not using `float` for anything, leverage `flex` instead.
2. Reference the wiki for CSS ordering and ensure your rules follow the guidelines.

File: src/index.html [hunk 7]
```diff
@@ -6,6 +6,7 @@
     <title></title>
     <meta name="description" content="">
     <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (andela-kakpobome) [inline src/index.html]: Since we are using custom fonts, what's the point of doing this?

File: src/components/AssetCategoryCard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import './AssetCategoryCard.scss';
+
+class AssetCategoryCard extends React.Component<{}> {
+  render() {
+    return (
+      <React.Fragment>
+        <div className="container-element">
+          <h2 className="asset-tag">ASSET CATEGORIES</h2>
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/index.tsx]: This should be in the `DashboardPage`. As you've rightly outlined, the name of this component is `AssetCategoryCard`, meaning it's just one (singular) card, it's not supposed to house all the cards, instead, the dashboard page should house the logic that creates as many of these cards as there are asset categories received from the API call.

File: src/assets/scss/globals.scss [hunk 9]
```diff
@@ -0,0 +1,10 @@
+@font-face {
+  font-family: 'Dinpro-regular';
+  src: url('../../assets/fonts/DIN.ttf'),
+    url('../../assets/fonts/din-webfont.woff'),
+    url('../../assets/fonts/din-webfont.woff2');
+}
+
+html {
+  background: rgb(246, 249, 252);
+}
```
Comment (andela-kakpobome) [inline src/assets/scss/globals.scss]: Let's have the `body` rule here, and also specify that `spans, divs, ps, section, h1, h2, h3, h4, h5...etc` should have `font-family` of DinPro, so you don't have to specify everytime henceforth

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 10]
```diff
@@ -0,0 +1,91 @@
+@import '../../assets/scss/variables';
+
+.grid-item {
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: It's easier to read when you nest related parent and children, in this context, `.room-tag` and `.item-value` should be nested in here.

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+  background-color: #fff;
+  border-radius: 6px;
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: We should only use relative measurements i.e `rem`. You can download `sass-rem` and simply wrap all your current measurements like so `rem(6px)`

File: src/assets/scss/globals.scss [hunk 12]
```diff
@@ -0,0 +1,10 @@
+@font-face {
+  font-family: 'Dinpro-regular';
+  src: url('../../assets/fonts/DIN.ttf'),
+    url('../../assets/fonts/din-webfont.woff'),
+    url('../../assets/fonts/din-webfont.woff2');
```
Comment (andela-kakpobome) [inline src/assets/scss/globals.scss]: We need to have several `@font-faces` that specify the appropriate weights for the font family. i.e 

```
@font-face {
   font-family: 'Dinpro-light';
   font-weight: 300;
   src: url('../../assets/fonts/DIN -light.ttf'),
   url('../../assets/fonts/din-webfon-lightt.woff'),
   url('../../assets/fonts/din-webfont-light.woff2');
}

@font-face {
   font-family: 'Dinpro-medium';
   font-weight: 500;
   src: url('../../assets/fonts/DIN -medium.ttf'),
   url('../../assets/fonts/din-webfon-medium.woff'),
   url('../../assets/fonts/din-webfont-medium.woff2');
}

@font-face {
   font-family: 'Dinpro-bold';
   font-weight: 700;
   src: url('../../assets/fonts/DIN-bold.ttf'),
   url('../../assets/fonts/din-webfon-bold.woff'),
   url('../../assets/fonts/din-webfont-bold.woff2');
}
```

Notice how I specified `font-weight` that match with either `light, medium, bold` versions of the font

File: config/webpack.config.js (hunk 1 above)
Comment (Billmike) [reply config/webpack.config.js]: Fixed.

File: config/webpack.dev.js (hunk 3 above)
Comment (Billmike) [reply config/webpack.dev.js]: Fixed.

File: src/components/AssetCategoryCard/index.tsx (hunk 8 above)
Comment (Billmike) [reply src/components/AssetCategoryCard/index.tsx]: Fixed.

File: src/assets/scss/globals.scss (hunk 9 above)
Comment (Billmike) [reply src/assets/scss/globals.scss]: Fixed.

File: src/components/AssetCategoryCard/AssetCategoryCard.scss (hunk 11 above)
Comment (Billmike) [reply src/components/AssetCategoryCard/AssetCategoryCard.scss]: Fixed

File: src/assets/scss/globals.scss (hunk 12 above)
Comment (Billmike) [reply src/assets/scss/globals.scss]: Fixed.

File: src/components/AssetCategoryCard/AssetCategoryCard.scss (hunk 10 above)
Comment (Billmike) [reply src/components/AssetCategoryCard/AssetCategoryCard.scss]: Fixed.

File: .gitignore [hunk 13]
```diff
@@ -9,3 +9,4 @@ yarn-error.log
 package-lock.json
 /reports
 src/**/**/__snapshots__/*.test.tsx.snap
+src/assets/fonts/DINRegular
```
[the remark this answers is not in the document]
Comment (andela-kakpobome) [reply .gitignore]: Why we don't just delete?

File: src/assets/scss/variables.scss [hunk 14]
```diff
@@ -0,0 +1,2 @@
+$andela-blue: #3359DB;
+$light-gray: #323c47;
```
Comment (andela-kakpobome) [inline src/assets/scss/variables.scss]: should be uppercase

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+  
+  .item-value {
+    margin-right: 1.5rem;
+    margin-top: rem(1px);
+    font-size: rem(48px);
+    text-align: right;
+    color: #333E44;
+  }
+}
+
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: Lose this line

File: src/pages/DashboardPage/DashboardPage.scss [hunk 16]
```diff
@@ -0,0 +1,51 @@
+@import '~sass-rem';
+@import '../../assets/scss/variables';
+
+.container-element {
```
Comment (andela-kakpobome) [inline src/pages/DashboardPage/DashboardPage.scss]: CSS rule ordering was not followed on this page.

File: src/pages/DashboardPage/DashboardPage.scss [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+    margin-left: 2.5rem;
+    padding-top: rem(16px);
+    font-size: 24px;
+    color: #333E44;
+  }
+  
+  .download-icon {
+    margin-left: rem(5px);
+  }
+
```
Comment (andela-kakpobome) [inline src/pages/DashboardPage/DashboardPage.scss]: Lose this line

File: src/components/AssetCategoryCard/index.tsx [hunk 18]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import './AssetCategoryCard.scss';
+
+class AssetCategoryCard extends React.Component<{}> {
+  render() {
+    return (
+      <React.Fragment>
+        <div className="l-wrap">
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/index.tsx]: Again, this component should return one card, and in Dashboard, or wherever it's used, you loop through and create as many of these cards as required.

File: src/pages/DashboardPage/DashboardPage.scss (hunk 17 above)
Comment (Billmike) [reply src/pages/DashboardPage/DashboardPage.scss]: Fixed.

File: src/assets/scss/variables.scss (hunk 14 above)
Comment (Billmike) [reply src/assets/scss/variables.scss]: Fixed.

File: .gitignore (hunk 13 above)
[the remark this answers is not in the document]
Comment (Billmike) [reply .gitignore]: Fixed.

**Comment by missating** [comment]
@Billmike Your commit message is not in the right format. For features, The header uses `feat` not 'feature` as oppose to `chore` for chores.

**Comment by Billmike** [comment]
@missating Fixed

File: src/pages/DashboardPage/DashboardPage.scss (hunk 16 above)
Comment (Billmike) [reply src/pages/DashboardPage/DashboardPage.scss]: Fixed.

File: src/components/AssetCategoryCard/interface.d.ts [hunk 19]
```diff
@@ -0,0 +1,4 @@
+export interface AssetCategoryCardProps {
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/interface.d.ts]: This file should be called `interfaces.d.ts`

File: src/components/AssetCategoryCard/index.tsx [hunk 20]
```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { AssetCategoryCardProps } from './interface';
+import './AssetCategoryCard.scss';
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/index.tsx]: Refer to the wiki on how to order imports, you should also include the `line comments` that clearly separate the different kinds of imports

File: src/components/AssetCategoryCard/AssetCategoryCard.test.tsx [hunk 21]
```diff
@@ -0,0 +1,16 @@
...
+import AssetCategoryCard from './index';
+
+describe('AssetCategoryCard component', () => {
+  it('should render correctly', () => {
+    const props = {
+      name: 'Mac',
+      count: 10
+    };
+    const wrapper = shallow(
+      <AssetCategoryCard name={props.name} count={props.count} />
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.test.tsx]: Shorten the prop assignment i.e `<AssetCategoryCard { ...props } />`

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 22]
```diff
@@ -0,0 +1,35 @@
...
+.grid-item {
+  flex-basis: 17.7%;
+  background-color: map-get($grey, 100);
+  box-shadow: rem(0 2px 9px 0 rgba(0, 0, 0, 0.05));
+  border-radius: 0.375rem;
+  height: 9.375rem;
+  margin-right: 2.3rem;
+  margin-top: 1.25rem;
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: The order here should be:

```
flex-basis: 17.7%;
background-color: map-get($grey, 100);
height: 9.375rem;
margin-right: 2.3rem;
margin-top: 1.25rem;
border-radius: 0.375rem;
box-shadow: rem(0 2px 9px 0 rgba(0, 0, 0, 0.05));
```

File: src/components/AssetCategoryCard/AssetCategoryCard.scss [hunk 23]
```diff
@@ -0,0 +1,35 @@
+@import '~sass-rem';
+@import '../../assets/scss/variables';
+
+.l-wrap {
```
Comment (andela-kakpobome) [inline src/components/AssetCategoryCard/AssetCategoryCard.scss]: We talked about giving this a better name already.

File: src/assets/scss/variables.scss [hunk 24]
```diff
@@ -0,0 +1,27 @@
+$light-gray: #323C47;
```
Comment (andela-kakpobome) [inline src/assets/scss/variables.scss]: This should be somewhere in the `$grey` map, there's no reason it's standing alone.

File: src/assets/scss/globals.scss [hunk 25]
```diff
@@ -0,0 +1,25 @@
+@font-face {
+  font-family: 'Dinpro-light';
```
Comment (andela-kakpobome) [inline src/assets/scss/globals.scss]: For this to work properly, you have to set the `font-family` to be the same value for all `font faces` i.e `DinPro`. With what you have you essentially defining 3 different font-familie, that's why it didn't work earlier.

File: src/assets/scss/globals.scss [hunk 26]
```diff
@@ -0,0 +1,25 @@
...
+  font-weight: 700;
+  src: url('../../assets/fonts/DINPRO-Bold.ttf');
+}
+
+html {
+  background: rgb(246, 249, 252);
+}
+
+* {
+  font-family: 'Dinpro-light';
```
Comment (andela-kakpobome) [inline src/assets/scss/globals.scss]: Refer to the comment above about having a unified font-family value.

File: src/assets/images/icon.svg [hunk 27]
```diff
@@ -0,0 +1,25 @@
...
+                                <use xlink:href="#path-1"></use>
+                            </mask>
+                            <use id="icon" fill="#BCBCBC" fill-rule="nonzero" xlink:href="#path-1"></use>
+                        </g>
+                    </g>
+                </g>
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (andela-kakpobome) [inline src/assets/images/icon.svg]: EoF line required here.

File: src/assets/scss/variables.scss (hunk 24 above)
Comment (Billmike) [reply src/assets/scss/variables.scss]: Fixed

File: src/assets/scss/globals.scss (hunk 26 above)
Comment (Billmike) [reply src/assets/scss/globals.scss]: Fixed

File: src/components/AssetCategoryCard/interface.d.ts (hunk 19 above)
Comment (Billmike) [reply src/components/AssetCategoryCard/interface.d.ts]: Fixed

File: src/assets/scss/globals.scss (hunk 25 above)
Comment (Billmike) [reply src/assets/scss/globals.scss]: Fixed

File: src/components/AssetCategoryCard/index.tsx (hunk 20 above)
Comment (Billmike) [reply src/components/AssetCategoryCard/index.tsx]: Fixed.

File: src/assets/images/icon.svg (hunk 27 above)
Comment (Billmike) [reply src/assets/images/icon.svg]: Fixed.

File: src/App.tsx [hunk 28]
```diff
@@ -2,14 +2,13 @@ import * as React from 'react';
 
 import AuthPage from 'pages/AuthPage';
 import Dashboard from 'pages/Dashboard';
```
Comment (andela-kakpobome) [inline src/App.tsx]: Missed this earlier, but this import should now be irrelevant, please remove.

File: src/pages/AuthPage/AuthPage.scss [hunk 29]
```diff
@@ -1,12 +1,6 @@
 $black: #000;
 $white: #fff;
```
Comment (andela-kakpobome) [inline src/pages/AuthPage/AuthPage.scss]: We can remove these now, and reference the palette instead.

File: src/pages/Dashboard/index.tsx [hunk 30]
```diff
@@ -6,23 +6,25 @@ export class Dashboard extends React.Component<DashboardProps, DashboardState> {
...
+      isLoadingCategories: true
     };
   }
 
   render() {
     return (
       <div>
         <h1>Categories</h1>
-          {this.props.categories.map(category => <div key="category.id">{category.name}</div>)}
+        {this.props.categories.map(category => (
```
Comment (andela-kakpobome) [inline src/pages/Dashboard/index.tsx]: Can you please have this instead:
```
{
   this.props.categories.map(category => (
      <div key="category.id">{category.name}</div>
   ))
}
```

Cleaner and somewhat easier to read, especially when we add more content here

File: src/pages/Dashboard/index.tsx [hunk 31]
```diff
@@ -6,23 +6,25 @@ export class Dashboard extends React.Component<DashboardProps, DashboardState> {
   constructor(props) {
```
Comment (andela-kakpobome) [inline src/pages/Dashboard/index.tsx]: Also implement the import ordering here.

File: src/pages/DashboardPage/DashboardPage.scss [hunk 32]
```diff
@@ -0,0 +1,54 @@
...
+    margin-right: rem(25px);
+    vertical-align: middle;
+    line-height: rem(36.8px);
+    text-align: center;
+    color: map-get($pallette, light);
+  }
+  
+  .export-asset {
+    border-radius: rem(2px);
+    box-shadow: rem(0 2px 9px 0 rgba(0, 0, 0, 0.03));
```
Comment (andela-kakpobome) [inline src/pages/DashboardPage/DashboardPage.scss]: 29 and 30, don't deserve to be in these positions

File: src/pages/DashboardPage/DashboardPage.scss [hunk 33]
```diff
@@ -0,0 +1,54 @@
...
+  width: 91.67vw;
+  margin: 0 auto;
+  margin-top: rem(120px);
+
+  .header-row {
+    display: flex;
+    align-items: center;
+  }
+
+  .add-asset {
```
Comment (andela-kakpobome) [inline src/pages/DashboardPage/DashboardPage.scss]: Why do we still have specific styling for these buttons when they are already styled in the `Button` component?

File: src/pages/DashboardPage/DashboardPage.scss [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+    padding-left: rem(5px);
+    border: 1px solid map-get($pallette, primary);
+    vertical-align: middle;
+    line-height: rem(25.6px);
+    text-align: center;
+    color: map-get($pallette, primary);
+  }
+  
+  .asset-tag {
+    font-size: 24px;
```
Comment (andela-kakpobome) [inline src/pages/DashboardPage/DashboardPage.scss]: Use `rem`

**Pull request merged**